File: c/is_prime.c
#include <stdio.h>
#include <stdbool.h>

int isprime(int n)
{
  if (n == 2){
    return true;
  }

  if (n % 2 == 0){
    return false;
  }

  for (int i = 3; i * i <= n; i += 2){
    if (n % i == 0){
      return false;
    }
  }

  return true;
}

int main(void)
{
  int n;
  printf("Enter a number: ");
  scanf("%d", &n);

  if (isprime(n)){
    printf("Prime\n");
  } else {
    printf("Not prime\n");
  }

  return 0;
}

File: python/is_prime.py
def is_prime(n):
    if n < 2:
        return False
    if n == 2:
        return True
    if n % 2 == 0:
        return False
    mid = int(n ** 0.5) + 1
    for i in range(3, mid, 2):
        if n % i == 0:
            return False
    return True

print(is_prime(2))
print(is_prime(3))
print(is_prime(4))